compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java

```diff
@@ -69,15 +69,20 @@ public enum DiagnosticsCodes {
     PERSIST_121("PERSIST_121", "unsupported features: {0} type is not supported",
             ERROR),
 
+    // todo: Add expression type in the diagnostic message
     PERSIST_201("PERSIST_201",
-            "Unsupported expression: read() function of persist client does not support this expression", ERROR),
+            "persist client read() function not supported for this expression", ERROR),
     PERSIST_202("PERSIST_202",
-            "Unsupported expression: read() function of persist client only supports int value in limit clause",
-            ERROR),
+            "persist client read() function limit not supported for variables, " +
+                    "limit expression should be an integer", ERROR),
```

Comment:
```suggestion
            "persist client read() function limit expression doesn't support for variable references, " +
                    "it should be an integer value", ERROR),
```

Reply:
Fixed in [4789703](https://github.com/ballerina-platform/module-ballerina-persist/pull/68/commits/4789703653ca02e339872b67b15944f2f061ab26)